### PR TITLE
Feature/334 bulk export

### LINKED ISSUE #334: Bulk export
The json files for the individual workflows should be packaged into a single zip file. Delivering a zip file as opposed to one large json file with an array of workflows allows the user to easier exclude workflows from the zip file in the import. The user might want to exclude workflows to try a few workflows before importing a big batch of workflows or exclude workflows that cause import errors (workflows that violate validations)

Action should complete fully or not at all.

### PR DESCRIPTION
**Pull request by kevinwallimann**
Closes #334

### CODE REVIEW

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala [hunk 1]
```diff
@@ -106,18 +109,28 @@ class WorkflowController @Inject()(workflowService: WorkflowService) {
...
-        ObjectMapperSingleton.getObjectMapper.writerWithDefaultPrettyPrinter.writeValueAsBytes(workflowExport)
-      )
-
-      ResponseEntity.ok()
-        .contentType(MediaType.parseMediaType("application/json"))
-        .header(HttpHeaders.CONTENT_DISPOSITION, s"attachment; filename=${workflowExport.workflowJoined.name}.json")
-        .body(resource)
-    }.toJava.toCompletableFuture
+  @GetMapping(path = Array("/workflows/export"))
+  def exportWorkflows(@RequestParam jobIds: Array[Int]): CompletableFuture[ResponseEntity[ByteArrayResource]] = {
```
Comment (jozefbakus) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala]: jobIds should be array of Long

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala [hunk 2]
```diff
@@ -106,18 +109,28 @@ class WorkflowController @Inject()(workflowService: WorkflowService) {
...
+    val jobIdsLong = jobIds.map(_.toLong).toSeq
+    if (jobIdsLong.size > 1) {
+      workflowService.exportWorkflowsAsZip(jobIdsLong).map(baos =>
+        ResponseEntity.ok()
+          .contentType(MediaType.parseMediaType("application/zip"))
+          .header(HttpHeaders.CONTENT_DISPOSITION, s"attachment; filename=workflows${environment}.zip")
```
Comment (jozefbakus) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala]: What about dash ? workflows-${environment}.zip

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala (hunk 1 above)
Comment (jozefbakus) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala]: I would create two endpoints, one for single workflow and one for multiple workflows.
Here is one function which does two things. 
But leave it as it is.

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowService.scala [hunk 3]
```diff
@@ -191,6 +196,26 @@ class WorkflowServiceImpl(override val workflowRepository: WorkflowRepository,
     } yield WorkflowImportExportWrapper(workflow, jobTemplates)
   }
 
+  override def exportWorkflowsAsZip(workflowIds: Seq[Long])(implicit ec: ExecutionContext): Future[ByteArrayOutputStream] = {
+    val workflowExportsFut = Future.sequence(workflowIds.map(exportWorkflow(_)))
```
Comment (jozefbakus) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowService.scala]: I would rather do it with one call to db. (or two, one for workflows and one for templates)

**Review by jozefbakus — CHANGES_REQUESTED** [review]

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala (hunk 2 above)
Comment (kevinwallimann) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala]: ok

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala (hunk 1 above)
Comment (kevinwallimann) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala]: I could extract the parts into two functions. I think two endpoints aren't necessary. If you would call `/workflows/export/42` you would get a json wrapped in a zip file, while `workflow/export/42` would return only the json. I think it doesn't make sense to have two ways of exporting.

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowService.scala (hunk 3 above)
Comment (kevinwallimann) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowService.scala]: ok

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala (hunk 1 above)
Comment (kevinwallimann) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala]: Ok.

Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala]: OK

**Review by jozefbakus — APPROVED** [review]
👍🏻

**Pull request merged**